tests/integration/tier0/non-destructive/els/test_els_support.py

```diff
@@ -0,0 +1,125 @@
+import os.path
+
+import pytest
+
+from conftest import TEST_VARS
+
+
+@pytest.fixture
+def els_mock(shell, backup_directory):
+    """
+    Fixture to mock different scenarios while handling the ELS candidates.
+    Backs up the convert2rhel/systeminfo.py and modifies the ELS_RELEASE_DATE variable.
+    Restores the original version after the test is done.
+    """
+    els_startdate_file = shell("find /usr -path '*/convert2rhel/systeminfo.py'").output.strip()
+    els_check_file = shell("find /usr -path '*/convert2rhel/actions/system_checks/els.py'").output.strip()
+    backup_startdate_file = os.path.join(backup_directory, "systeminfo.py.bkp")
+    backup_check_file = os.path.join(backup_directory, "els.py.bkp")
+    file_backup_mapping = {els_startdate_file: backup_startdate_file, els_check_file: backup_check_file}
+    for file, backup in file_backup_mapping.items():
+        assert shell(f"cp {file} {backup}").returncode == 0
+
+    original_startdate_value = '"2024-06-12"'
+    original_releasever_value = "if system_info.version.major == 7:"
+
+    def _update_els_mapping(modified_startdate=None, modified_releasever=None):
+        if modified_startdate:
+            shell(f"sed -i 's/{original_startdate_value}/{modified_startdate}/' {els_startdate_file}")
+        if modified_releasever:
+            shell(f"sed -i 's/{original_releasever_value}/{modified_releasever}/' {els_check_file}")
+
+    yield _update_els_mapping
+
+    def _restore_els_mapping():
+        for file, backup in file_backup_mapping.items():
+            assert shell(f"mv -v {backup} {file}").returncode == 0
+
+    _restore_els_mapping()
+
+
+REGULAR_REPOID_MSG = "RHEL repository IDs to enable: rhel-7-server-rpms"
+ELS_REPOID_MSG = "RHEL repository IDs to enable: rhel-7-server-els-rpms"
+
+MOCKED_ELS_PHASE = '"1970-01-01"'
+MOCKED_NON_ELS_SYSVER = "if system_info.version.major == 42:"
+
+
+els_support_parameters = [
+    # System not recognized as an ELS candidate with --els option used
+    ("--els", REGULAR_REPOID_MSG, None, MOCKED_NON_ELS_SYSVER, False, False),
+    # System recognized as an ELS candidate after the start of ELS phase without --els option
+    ("", REGULAR_REPOID_MSG, MOCKED_ELS_PHASE, None, False, True),
+]
+
+
+@pytest.mark.parametrize(
+    "additional_option, repoid_message, modified_startdate, modified_releasever, is_els, recommend_els_msg_displayed",
```

Comment:
Do we need the `is_els` parameter if there is no test case where it is set to `True`?